### PR TITLE
Update dependencies, regen and fix cbor-gen output, add dependabot

### PR DESCRIPTION
**Pull request by rvagg**
This one's a little more tricky because we don't use the raw cbor-gen output but do some modifications here to deal with a custom Pointer type (pointer_cbor.go isn't included in cbor-gen generate but I manually made a minor edit) and the custom big.Int handling here. So the changes here are a unification of the new cbor-gen output and the modified one.

### CODE REVIEW

File: cbor_gen.go [hunk 1]
```diff
@@ -1,5 +1,3 @@
-// Code generated by github.com/whyrusleeping/cbor-gen. DO NOT EDIT.
```
Comment (Stebalien) [inline cbor_gen.go:1]: If this isn't generated, we shouldn't call it `cbor_gen.go`. More generally, why isn't this generated?

Comment (Stebalien) [reply cbor_gen.go:1]: I see. I'm going to fix this the right way.

Comment (Stebalien) [reply cbor_gen.go:1]: https://github.com/filecoin-project/go-hamt-ipld/pull/124

**Comment by rvagg** [comment]
I tried this branch out on an nv23 migration and it got the same state root as the currently released version, so there's some verification that we're not doing anything funky here.

**Comment by rvagg** [comment]
@Stebalien updated this, care to take another look and we can land it?

**Review by Stebalien — APPROVED** [review]

**Comment by Stebalien** [comment]
:heart:

**Pull request merged**